node/nodeTesting.go

```diff
@@ -181,12 +184,13 @@ func (n *Node) generateAndSignSingleTx(
 	rcvAddrBytes []byte,
 	sndAddrBytes []byte,
 	data string,
+	sk crypto.PrivateKey,
 ) (*transaction.Transaction, []byte, error) {
 
 	if n.marshalizer == nil {
```

Comment:
check if nil here and the following lines

Reply:
done.